### PR TITLE
Internal: Update open_social_dev to 2.1

### PR DESCRIPTION
**Pull request by Kingdutch**
## Problem / Solution
The 2.0 version of the package should've been needed for Drupal 10 compatibility but apparently in the distribution that works fine with 1.4.0.

Nevertheless we should upgrade to 2.0.0 to make sure we update Drush and all of our tooling.

We immediately jump to 2.1.0 because it adds `kingdutch/cucumber-linter` which we want to use for linting our Behat tests.

## Issue tracker
https://getopensocial.atlassian.net/browse/PROD-27811

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

File: modules/social_features/social_user/social_user.oauth2_scopes.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
+"viewer":
```
Comment (ronaldtebrake) [inline modules/social_features/social_user/social_user.oauth2_scopes.yml:1]: Hi @Kingdutch,

Could you maybe elaborate why this was added? 🤔

Comment (Kingdutch) [reply modules/social_features/social_user/social_user.oauth2_scopes.yml:1]: @ronaldtebrake Looks like I accidentally did `git add modules` while this was an untracked file in my workdir and it got added to this PR 😅 

The file should be removed from the PR (I've made a local copy of it)

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3625-60ftbxqlztkorja1lnponjajk19qjbvk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6566f8d50a5d5302cc90a20b

**Comment by ronaldtebrake** [comment]
Also in 12.1.x 🍒

**Pull request merged**